### PR TITLE
General minor tidy-up and update of docs

### PR DESCRIPTION
**Pull request by chrisseaton**

### CODE REVIEW

File: doc/contributor/make-distribution.md [hunk 1]
```diff
@@ -5,7 +5,7 @@ See the [Distributions page](../user/distribution.md).
 The tool to build these distributions is `tool/make-distribution.sh`.
 
 This script should be run in a fresh checkout of truffleruby in a new workspace.
-This is necessary as otherwise extra gems and build artifacts might be included.
+This is necessary as otherwise extra gems and build artefacts might be included.
```
Comment (nirvdrum) [inline doc/contributor/make-distribution.md]: This gets back to American vs British spelling.

File: doc/contributor/quick-rails-testing.md [hunk 2]
```diff
@@ -1,14 +1,14 @@
...
-When executed it prepares its environment, starts the Rails server, 
-runs a few tests against JSON API, and shut downs the rails server.
-It requires Redis as a database to be running in its default configuration on `localhost:6379`.
-
-The `jt test ecosystem rails-app` should be executed in TruffleRuby's home. 
-The output will be similar to following: 
+This repository contains a small Rails application for testing  in the
+`test/truffle/ecosystem/rails-app` directory.  The test can be executed with `jt
+test ecosystem rails-app` in TruffleRuby's home directory. When executed it
+prepares its environment, starts the Rails server,  runs a few tests against
```
Comment (nirvdrum) [inline doc/contributor/quick-rails-testing.md]: There's an extra space before "runs".

File: doc/contributor/workarounds.md [hunk 3]
```diff
@@ -1,37 +1,36 @@
...
-The patching file is responsible for loading the original file (if desirable),
-which can be done with `Truffle::Patching.require_original __FILE__`.
+The patching system is loaded in `post-boot.rb` and is able to patch stdlib and
+installed gems.  The code implementing the patching itself can be found in
+`lib/truffle/truffle/patching.rb`. It can be disabled by passing the
+`-Xpatching=false` option.
+
+The patching system works for gems as follows.  When a gem *g* is activated,
+there is a directory `g` in `lib/patches`,  and the directory is listed in
+`lib/truffle/truffle/patching.rb`,   the directory is inserted into `$LOAD_PATH`
```
Comment (nirvdrum) [inline doc/contributor/workarounds.md]: There's an extra space before "the directory."

File: doc/contributor/workarounds.md [hunk 4]
```diff
@@ -1,37 +1,36 @@
...
+ecosystem` mainly for Rails testing. The tool should be removed and no new tests
+should use it.
 
 ## Constant replacement
 
-`org.truffleruby.parser.ConstantReplacer` replaces constant 
-if a partial file path matches and the name of the constant equals.
-Usually used to fix Ruby implementation detection.
-Should be removed and not used further.    
+`org.truffleruby.parser.ConstantReplacer` replaces constant  if a partial file
```
Comment (nirvdrum) [inline doc/contributor/workarounds.md]: There's an extra space before "if."

File: doc/user/faq.md [hunk 5]
```diff
@@ -30,19 +31,30 @@ directly control the compiler. However this is complicated, so normally Truffle
...
+framework including JavaScript, Python and R.
+
 ### How do I get TruffleRuby?
 
-The easiest way to get a JVM with Graal is via the GraalVM, available from the
-Oracle Technology Network. This includes the JVM, the Graal compiler, and
-TruffleRuby, all in one package with compatible versions. You can get either the
-runtime environment (RE) or development kit (DK).
+The easiest way to get TruffleRuby is the GraalVM, available from the Oracle
+Technology Network. This includes the JVM, the Graal compiler, and TruffleRuby,
```
Comment (nirvdrum) [inline doc/user/faq.md]: I think we should link to the OTN download page here.

File: doc/user/truffleruby-additions.md [hunk 6]
```diff
@@ -38,6 +38,8 @@ used.
 `Truffle.native?` reports if TruffleRuby has been ahead-of-time compiled.
 In practice this implies that the SubstrateVM is being used.
 
+`Truffle.revision` reports the source control revision of TruffleRuby.
```
Comment (eregon) [inline doc/user/truffleruby-additions.md]: `reports the source control revision used to build TruffleRuby.` maybe?

File: doc/contributor/make-distribution.md (hunk 1 above)
Comment (chrisseaton) [reply doc/contributor/make-distribution.md]: Fixing.

File: doc/contributor/quick-rails-testing.md (hunk 2 above)
Comment (chrisseaton) [reply doc/contributor/quick-rails-testing.md]: Fixing.

File: doc/contributor/workarounds.md (hunk 3 above)
Comment (chrisseaton) [reply doc/contributor/workarounds.md]: Fixing.

File: doc/contributor/workarounds.md (hunk 4 above)
Comment (chrisseaton) [reply doc/contributor/workarounds.md]: Fixing.

File: doc/user/faq.md (hunk 5 above)
Comment (chrisseaton) [reply doc/user/faq.md]: Fixing.

File: doc/user/truffleruby-additions.md (hunk 6 above)
Comment (chrisseaton) [reply doc/user/truffleruby-additions.md]: Fixing.

**Pull request merged**